server/python/src/routers/movies.py

```diff
@@ -167,6 +240,361 @@ async def create_movies_batch(movies: List[CreateMovieRequest]):
 # Place find_and_delete_movie endpoint here
 #------------------------------------
 
+"""
+    DELETE /api/movies/{id}/find-and-delete
+    Finds and deletes a movie in a single atomic operation.
+    Demonstrates the findOneAndDelete() operation.
+    Path Parameters:
+        id (str): The ObjectId of the movie to find and delete.
+    Returns:
+        SuccessResponse[Movie]: A response object containing the deleted movie data.
+"""
+
+@router.delete("/{id}/find-and-delete", response_model=SuccessResponse[Movie])
+async def find_and_delete_movie(id: str):
+    object_id = ObjectId(id)
+
+    # Use find_one_and_delete() to find and delete in a single atomic operation
+    # This is useful when you need to return the deleted document
+    # or ensure the document exists before deletion
+    deleted_movie = await db.movies.find_one_and_delete({"_id": object_id})
+
+    if deleted_movie is None:
+        raise HTTPException(status_code=404, detail="Movie not found")
+
+    deleted_movie["_id"] = str(deleted_movie["_id"]) # Convert ObjectId to string
+    
+    return create_success_response(deleted_movie, "Movie found and deleted successfully")
+
+async def execute_aggregation(pipeline: list) -> list:
+    """Helper function to execute aggregation pipeline and return results"""
+    print(f"Executing pipeline: {pipeline}")  # Debug logging
+    print(f"Database name: {db.name if hasattr(db, 'name') else 'unknown'}")
+    print(f"Collection name: movies")
+    
+    # For motor (async MongoDB driver), we need to await the aggregate call
```

Comment:
I thought we weren't using Motor (instead using PyMongo's native async) 
```suggestion
    # For async PyMongo driver, we need to await the aggregate call
```